File: prog.cc
#include <mysql.h>
#include <iostream>
#include <string>

using namespace std;
MYSQL *conn, mysql;
MYSQL_RES *res;
MYSQL_ROW row;

int query_state;

int main()
{
    const char* server = "Courses";
    const char* user = "z1803493;
    const char* password = "1995Jul";
    const char* database = "z1803493;

    mysql_init(&mysql);
    conn = mysql_real_connect(&mysql, server, user, password, database, 0, 0, 0);

    if (!conn) {
        cout << mysql_error(&mysql) << endl;
        return -1;
    }
//item vs package
    string select_query = "Select i.item_id, p.package_id      from Item i, Package p      where i.item_id = p.item_id group by p.package_id ";
    query_state = mysql_query(conn, select_query.c_str());
    if (query_state) {
        cout << mysql_error(conn) << endl;
        return 1;
    }

    res = mysql_store_result(conn);
   
    cout<<"Item vs package"<<endl<<endl;
    while ((row = mysql_fetch_row(res)) != NULL) {
        cout<<"Item : "<<row[0]<<" Package id : "<<row[1]<<endl;
    }
    cout<<endl;

//vounteer cs task
     select_query = "Select v. vname  , t.task_description  from Volunteer v, task_assigned a, Task t  where v. volunteer_id = a.volunteer_id and a.tcode = t.tcode";
    query_state = mysql_query(conn, select_query.c_str());
    if (query_state) {
        cout << mysql_error(conn) << endl;
        return 1;
    }

    res = mysql_store_result(conn);
   
    cout<<"volunteer vs task "<<endl<<endl;
    while ((row = mysql_fetch_row(res)) != NULL) {
        cout<<"Volunteer name : "<<row[0]<<endl<<"Task Description : "<<row[1]<<endl;
    }
    cout<<endl;

//volunteer vs packing

    select_query = "Select v. vname, p.name_packaging from Volunteer v, task_assigned a, task_withshipping w, packaging_task p     where v. volunteer_id = a.volunteer_id and a.tcode = w.tcode and w.name_packaging = p.name_packaging";
    query_state = mysql_query(conn, select_query.c_str());
    if (query_state) {
        cout << mysql_error(conn) << endl;
        return 1;
    }

    res = mysql_store_result(conn);
   
    cout<<"volunteer vs packaging"<<endl<<endl;
    while ((row = mysql_fetch_row(res)) != NULL) {
        cout<<"Volunteer Name : "<<row[0]<<endl<<"Packaging Name : "<<row[1]<<endl;
    }
    cout<<endl;

//item vs count
    select_query = "Select i.item_description, count(i.item_id)     from Item i, Package p , packaging_task t  where i.item_id = p.item_id and p.package_id = t.package_id group by i.item_id";
    query_state = mysql_query(conn, select_query.c_str());
    if (query_state) {
        cout << mysql_error(conn) << endl;
        return 1;
    }

    res = mysql_store_result(conn);
   
    cout<<"Item vs count"<<endl<<endl;
    while ((row = mysql_fetch_row(res)) != NULL) {
        cout<<"Item : "<<row[0]<<endl<<"Count : "<<row[1]<<endl;
    }
    cout<<endl;

//list vs weight

    select_query = "Select  distinct l.list_name , p.weight from Packaging_list l ,Item i ,Package p      where l.item_id = i.item_id and p.item_id = i.item_id";
    query_state = mysql_query(conn, select_query.c_str());
    if (query_state) {
        cout << mysql_error(conn) << endl;
        return 1;
    }

    res = mysql_store_result(conn);
   
    cout<<"List vs weight"<<endl<<endl;
    while ((row = mysql_fetch_row(res)) != NULL) {
        cout<<"list : "<<row[0]<<endl<<"weight : "<<row[1]<<endl;
    }
    cout<<endl;


//updating volunteer names

    select_query = " update Volunteer set vname = \'SRK\' where volunteer_id = \'3\'";
    query_state = mysql_query(conn, select_query.c_str());
    if (query_state) {
        cout << mysql_error(conn) << endl;
        return 1;
    }
    cout<<"Updated the volunteer name coulumn"<<endl;

//deleting item
    select_query = "delete from Item where value = 0.01";
    query_state = mysql_query(conn, select_query.c_str());
    if (query_state) {
        cout << mysql_error(conn) << endl;
        return 1;
    }
    cout<<"Deletion sucessfully"<<endl;
//Updating item value
    select_query = "Update Item set value = 8.0 where value = 0.01";
    query_state = mysql_query(conn, select_query.c_str());
    if (query_state) {
        cout << mysql_error(conn) << endl;
        return 1;
    }
    cout<<"Updated the Item value"<<endl;

    
    mysql_free_result(res);
    mysql_close(conn);
}

File: prog.py
 

import MySQLdb

db = MySQLdb.connect("localhost","root","4451","group_project" )

query = db.cursor()

###Show stats for start time 
q = """select start_time from   `assigned_to`  procedure analyse ( ) """
query.execute(q)
for data in query:
	print "First packing was done at  - ",data[1]
	print "Last pacing was done at  - ",data[2]
	
###show stats for end time
q = """select end_time from   `assigned_to`  procedure analyse ( ) """
query.execute(q)
for data in query:
	print "First package was packed at   - ",data[1]
	print "Last package was packed at  - ",data[2]
	print "Average Packing time - ",data[7]

###show stats for package data
q = """select packeageDate from   `package`  procedure analyse ( )"""
query.execute(q)
for data in query:
	print "First package was packed on " , data[1]
	print "Last package was packed on " , data[2]
	print "Average inverval for packign packages is  ", data[7]," days"

###show stats for weight
q = """select weight from   `package`  procedure analyse ( )"""
query.execute(q)
for data in query:
	print "Max weight of a package is  " , data[1]
	print "Least weight of a package is " , data[2]
	print "Average weight of the package is  ", data[7]

###list all the organizatio alons with list
q = """select concat(name,", ",address) as Organization from united_helper"""

query.execute(q)
print "List of Organization : "
for data in query.fetchall():
	for i in data:
		print " ",i

###for volunteer and corresponding package
q = """select volunteer.name as vname, packing_list.name as pname from
	assigned_to join volunteer on 
	volunteer.volunteerId = assigned_to.volunteerId  join task on 
	task.code  = assigned_to.code join packing_task on 
	task.code  = packing_task.code join packing_list on 
	packing_list.listId = packing_task.listId"""
query.execute(q)
print "Volunteer Handling which package : "
for data in query.fetchall():
	print " ",data[0]," - ",data[1]
q = """select package.packageId,item.name from package join contains on
	package.packageId = contains.packageId join item on 
	contains.itemId = item.itemId"""
query.execute(q)
print " which Package contains which items: "
for data in query.fetchall():
	print " ",data[0]," - ",data[1]

###packge with corresponsind task
q = """ select item.name , packing_task.totalPackages from package 
	join contains on package.packageId = contains.packageId 
	join item on  contains.itemId = item.itemId 
	join packing_task on package.code = packing_task.ptask_code"""
query.execute(q)
print " Count of packages in each packing task: "
for data in query.fetchall():
	print " ",data[0]," - ",data[1]

###addition of package alon with error handling
print """Creating new package : """
insert_packingid = raw_input("Enter package id :")
insert_listid  = raw_input("Enter packing list id : ");
insert_taskcode  = raw_input("Enter packing task code :");
insert_weight  = raw_input("Enter weight of the package : ");
insert_packing_date  = raw_input("Enter the date of packaging : ");
q = "insert into package values ("+insert_packingid+","+insert_weight+",'"+insert_packing_date+"',"+insert_taskcode+","+insert_listid+")"
try:
	query.execute(q)
	print "Update a Package Sucessfully !"
except MySQLdb.Error, e:
	print "error incorrect package,list id or task code"


###deleting package incuidng error handling
delete_pid = raw_input("Enter the package id to delete :")
q = "delete from package where packageId = " + delete_pid 
p = "delete from contains where packageId = "+delete_pid
query.execute(q)
try:
	query.execute(p)
except MySQLdb.Error, e:
	print "package id not exists"

###updating Volunteer details
vid = raw_input("Enter the Volunteer id to change : ")
vname = raw_input("Enter the volunteer name : ")
vaddress = raw_input("Enter volunteer address : ")
vphone = raw_input("Enter the phone number : ")
q = "update volunteer set name = '"+vname+"',address = '"+vaddress+"', phoneNo = '"+vphone+"' where volunteerId = " +vid

try:
	query.execute(q)
	print " Sucessfully updated !"
except MySQLdb.Error, e:
	print "Invalid Volunteer Id"


### Addition of new organization
oid = raw_input("Enter Organization id :") 
oname = raw_input("Enter new organization Name : ")
oaddress = raw_input("Enter  address : ") 
ophone = raw_input("Enter phone : ")

q = "insert into united_helper values ("+oid+",'"+oname+"','"+oaddress+"','"+ophone+"')";
query.execute(q)
print "updated "

db.close()
